go/store/prolly/tree/merge.go

```diff
@@ -51,12 +51,14 @@ func ThreeWayMerge[K ~[]byte, O Ordering[K], S message.Serializer](
 	serializer S,
 ) (final Node, stats MergeStats, err error) {
 
-	ld, err := DifferFromRoots[K](ctx, ns, ns, base, left, order)
+	// TODO: This function is only ever called to merge artifacts. Is is possible for the tuple desc to change here?
+	// If it does, do we want every row to appear in the diff?
```

Comment:
You're right that this code is only ever currently invoked when merging artifacts. 

You might consider passing in a boolean param to wire through to the calls to `DifferFromRoots`. That way, at least the caller would be being explicit about what behavior they wanted. The spot using this for merging artifacts could safely pass `false`, since the tuple description won't change for artifacts, as far as I know. 

Would be good to resolve this `TODO` before merging in though.